tests/resolve_imports/file_parser.cpp

```diff
@@ -0,0 +1,209 @@
+/*
+Copyright libCellML Contributors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "gtest/gtest.h"
+
+#include <algorithm>
+#include <fstream>
+#include <iostream>
+#include <libcellml>
+#include <sstream>
+#include <vector>
+
+// generated with test resource locations
+#include "test_resources.h"
+
+static std::string resolvePath(const std::string& filename, const std::string& base)
+{
+    // we can be naive here as we know what we are dealing with
+    std::string path = base.substr(0, base.find_last_of('/')+1) + filename;
+    return path;
+}
+
+static size_t countImportedChildren(libcellml::ComponentPtr parent)
+{
+    size_t numberImportedChildren = 0;
+    for (size_t n = 0; n < parent->componentCount();  ++n)
+    {
+        libcellml::ComponentPtr c = parent->getComponent(n);
+        if (c->isImport()) ++numberImportedChildren;
+        numberImportedChildren += countImportedChildren(c);
+    }
+    return numberImportedChildren;
+}
+
+static size_t countUnresolvedC(libcellml::ComponentPtr component);
+static size_t countUnresolvedComponents(libcellml::ComponentPtr component);
+static size_t countUnresolvedImportedComponents(libcellml::ModelPtr model);
+
+static size_t countUnresolvedC(libcellml::ComponentPtr component)
```

Comment:
Rename to `countUnresolvedComponent()` (i.e. similar for `resolveComponent()` below).

Actually, I would probably have `doCountUnresolvedImportedComponent()`, `doCountUnresolvedImportedComponents()` and `countUnresolvedImportedComponents()` (and something similar for the resolve methods below). Indeed, I originally got confused when I first saw `countUnresolvedC()` and `countUnresolvedComponents()` and then `countUnresolvedImportedComponents()`. For a second, I thought that the first two methods were not dealing with imported components...